packages/core/src/bar-chart.ts

```diff
@@ -106,7 +106,11 @@ export class BarChart extends BaseAxisChart {
 						.attr("fill", d => this.getFillScale()[d.datasetLabel](d.label))
 						.attr("stroke", d => this.options.accessibility ? this.colorScale[d.datasetLabel](d.label) : null)
 						.attr("stroke-width", Configuration.bars.default.strokeWidth)
-						.attr("stroke-opacity", d => this.options.accessibility ? 1 : 0);
+						.attr("stroke-opacity", d => this.options.accessibility ? 1 : 0)
+						// Make the data element accessible and pass x and y values for the data point, and the dataset label
+						.call(this.makeAccessible,
+							(d, index) => `Label: ${d.label}, Value: ${d.value}, belongs to ${d.datasetLabel}`,
+							d => this.y(Math.max(0, d.value)));
```

Comment:
the above 3 lines are very hard to read/understand. Only `makeAccessible` should be called here, and `makeAccessible` should know what to do with the data passed in.

If separate logic is needed for `bar-chart`, then `makeAccessible` should be overriden in `bar-chart`.

Reply:
Just passing a string containing the label, value, and dataset into makeAccessible Separate logic isn't needed for bar chart, every chart will also be passing in a string with those values.

Reply:
Nevermind, I understand after seeing your comment below

Reply:
Ok great. This function is still hard to read though. If you just call `this.makeAccesible`, `d` will be passed into it, and you can do all of this in there

Reply:
ok